### PR TITLE
Start in set mode

### PR DESCRIPTION
**Pull request by dasmaetthes**
Added settings to set the view mode for start(edit mode or markdown
mode)

### CODE REVIEW

File: src/mainwindow.cpp [hunk 1]
```diff
@@ -146,7 +146,7 @@ MainWindow::MainWindow(QWidget *parent) :
     ui->encryptedNoteTextEdit->setTabStopWidth(width);
 
     // set the edit mode for the note text edit
-    this->setNoteTextEditMode(true);
+    //this->setNoteTextEditMode(true);
```
Comment (pbek) [inline src/mainwindow.cpp]: why did you have to remove this?

File: src/dialogs/settingsdialog.cpp [hunk 2]
```diff
@@ -164,10 +166,10 @@ void SettingsDialog::readSettings() {
             settings.value("notifyAllExternalModifications").toBool());
     ui->noteSaveIntervalTime->setValue(
             settings.value("noteSaveIntervalTime").toInt());
-    ui->toolbarIconSizeSpinBox->setValue(
```
Comment (pbek) [inline src/dialogs/settingsdialog.cpp]: why did you have to remove these 2 lines?

File: src/mainwindow.h [hunk 3]
```diff
@@ -172,6 +172,7 @@ private slots:
     SimpleCrypt crypto;
     QDateTime currentNoteLastEdited;
     bool notifyAllExternalModifications;
+    bool setViewMode;
```
Comment (pbek) [inline src/mainwindow.h]: I think `set` isn't a good prefix for a variable. Just `viewMode` might be better...

File: src/dialogs/settingsdialog.cpp (hunk 2 above)
Comment (dasmaetthes) [inline src/dialogs/settingsdialog.cpp]: Sorry I just used it for test case to understand better how it works and forgot to reverse it

File: src/mainwindow.cpp (hunk 1 above)
Comment (pbek) [inline src/mainwindow.cpp]: Does editing of encrypted notes still work with that?

Comment (dasmaetthes) [inline src/mainwindow.cpp]: As `setNoteTextEditMode()` is set now in `readSettingsFromSettingsDialog()`

File: src/dialogs/settingsdialog.cpp [hunk 4]
```diff
@@ -109,6 +109,8 @@ void SettingsDialog::storeSettings() {
                       QString::number(ui->toolbarIconSizeSpinBox->value()));
     settings.setValue("MainWindow/showRecentNoteFolderInMainArea",
                       ui->showRecentNoteFolderCheckBox->isChecked());
+    settings.setValue("MainWindow/setViewMode",
+                      ui->setViewMode->isChecked());
```
Comment (pbek) [inline src/dialogs/settingsdialog.cpp]: can you please name this checkbox something like `defaultViewModeCheckBox`

File: src/mainwindow.cpp (hunk 1 above)
Comment (dasmaetthes) [inline src/mainwindow.cpp]: Yes encrypted editing still works

**Pull request merged**